### DISCUSSION TITLE
Updating lines in pyplot

### DISCUSSION THREAD

**Discussion by jandegier** [zauberzeug/nicegui#844] [Q&A] [answered]
### Question

Hello, I started using NiceGUI last week, and I am very enthousiastic about it: great stuff!

Currently, I am a bit stuck and I am hoping someone can provide help. I have a matplotlib figure that contains a lot of data, and that requires updating some of that data based on new information that is received over time. What I typically do, instead of completely redrawing the figure, is something along those lines:
```python
import matplotlib.pyplot as plt
from random import random
import time

plt.ion()

fig, ax = plt.subplots()
line, = plt.plot([0, 1], [0, 1])
plt.show()

for _ in range(10):
    time.sleep(1)
    x, y = list(line.get_xdata()), list(line.get_ydata())
    # include newly received information
    line.set_xdata(x + [random()])
    line.set_ydata(y + [random()])
    # make sure figure is updated
    fig.canvas.draw()
    fig.canvas.flush_events()
```

Is it possible to update the plot also while using `ui.pyplot()`? I tried the following:

```python
from matplotlib import pyplot as plt
from nicegui import ui
from random import random

pyplot = ui.pyplot(figsize=(6, 5), close=False)
with pyplot:
    line, = plt.plot([0, 1], [0, 1])

def update_line():
    x, y = list(line.get_xdata()), list(line.get_ydata())
    # include newly received information
    line.set_xdata(x + [random()])
    line.set_ydata(y + [random()])
    # print(x)
    # make sure figure is updated
    pyplot.fig.canvas.draw()
    pyplot.fig.canvas.flush_events()
    # pyplot.update()

ui.timer(1, update_line)

ui.run()
```

But this does not result in an updated figure. If I print the list `x` within the `update_line` function, I see that it grows over time, as expected.

Is the approach I describe here feasible? How can I update the figure?

**Comment by jandegier** [comment, marked as answer]
Ah, my initial search did not result in similar questions, but I found that this question is already answered here: https://github.com/zauberzeug/nicegui/discussions/55

**Reply by falkoschindler** [reply]
For the record:
```py
    with pyplot:
        pyplot.fig.canvas.draw()
        pyplot.fig.canvas.flush_events()
```
